subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/internal/configurations/renderer/ConsoleConfigurationReportRenderer.java

```diff
@@ -0,0 +1,452 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.tasks.diagnostics.internal.configurations.renderer;
+
+import org.apache.commons.lang.StringUtils;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ConfigurationReportModel;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ReportArtifact;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ReportAttribute;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ReportCapability;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ReportConfiguration;
+import org.gradle.api.tasks.diagnostics.internal.configurations.model.ReportSecondaryVariant;
+import org.gradle.api.tasks.diagnostics.internal.configurations.spec.AbstractConfigurationReportSpec;
+import org.gradle.internal.logging.text.StyledTextOutput;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+
+/**
+ * The {@link AbstractConfigurationReportRenderer} extension that can be used to render a {@link ConfigurationReportModel}
+ * to the console with richly formatted output.
+ */
+public final class ConsoleConfigurationReportRenderer extends AbstractConfigurationReportRenderer<StyledTextOutput> {
+    @Nullable private StyledTextOutput output;
+    private int depth;
+    private boolean recursiveExtensionsPrinted;
+
+    public ConsoleConfigurationReportRenderer(AbstractConfigurationReportSpec spec) {
+        super(spec);
+    }
+
+    @Override
+    public void render(ConfigurationReportModel data, StyledTextOutput output) {
+        this.depth = 0;
+        this.output = output;
+        this.recursiveExtensionsPrinted = false;
+
+        final boolean hasAnyRelevantConfigs = data.getAllConfigs().stream().anyMatch(c -> spec.isPurelyCorrectType(c) || c.isLegacy());
+        if (hasAnyRelevantConfigs) {
+            if (spec.isSearchForSpecificVariant()) {
+                writeSearchResults(spec, data);
```

Comment:
I don't think you need to pass `spec` into all these functions, right?

Reply:
It wasn't saved as a field before the `ReportRenderer` refactor.  I'll clean this up now.